src/osvr/Client/GestureRemoteFactory.cpp

```diff
@@ -0,0 +1,161 @@
+/** @file
+    @brief Implementation
+
+    @date 2015
+
+    @author
+    Sensics, Inc.
+    <http://sensics.com/osvr>
+
+*/
+
+// Copyright 2015 Sensics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Internal Includes
+#include "GestureRemoteFactory.h"
+#include "VRPNConnectionCollection.h"
+#include <osvr/Common/ClientContext.h>
+#include <osvr/Common/ClientInterface.h>
+#include <osvr/Common/PathTreeFull.h>
+#include <osvr/Util/ChannelCountC.h>
+#include <osvr/Util/UniquePtr.h>
+#include <osvr/Common/OriginalSource.h>
+#include "InterfaceTree.h"
+#include <osvr/Util/Verbosity.h>
+#include <osvr/Common/CreateDevice.h>
+#include <osvr/Common/GestureComponent.h>
+#include <osvr/Common/RegisteredStringMap.h>
+
+// Library/third-party includes
+// - none
+
+// Standard includes
+// - none
+
+namespace osvr {
+namespace client {
+
+    class NetworkGestureRemoteHandler : public RemoteHandler {
+      public:
+        NetworkGestureRemoteHandler(vrpn_ConnectionPtr const &conn,
+                                    std::string const &deviceName,
+                                    boost::optional<OSVR_ChannelCount> sensor,
+                                    common::InterfaceList &ifaces)
+            : m_dev(common::createClientDevice(deviceName, conn)),
+              m_interfaces(ifaces), m_all(!sensor.is_initialized()),
+              m_sensor(sensor) {
+
+            auto gesture = common::GestureComponent::create();
+            m_dev->addComponent(gesture);
+
+            gesture->registerGestureHandler(
+                [&](common::GestureData const &data,
+                    util::time::TimeValue const &timestamp) {
+                    m_handleGesture(data, timestamp);
+                });
+            gesture->registerGestureMapHandler(
```

Comment:
I'd prefer all the map handling to take place in the Component, to keep the logic centralized.